src/components/Button/src/button.module.css

```diff
@@ -0,0 +1,94 @@
+.denhaag-button {
+  font-family: var(--denhaag-font-family);
+  border-radius: var(--denhaag-border-radius);
```

Comment:
@bddjong For these settings you could introduce design tokens that are common across all form inputs, instead of the generic `font-family` design token from your brand. For example:

```css
:root {
  --denhaag-form-input-font-family: var(--denhaag-font-family);
  --denhaag-form-input-border-radius: var(--denhaag-border-radius);
}
```

```css
.denhaag-button {
  font-family: var(--denhaag-form-input-font-family);
  border-radius: var(--denhaag-form-input-border-radius);
}
```

You could consider "form-input" an 'abstract' component, like the parent component of the button and text input components. You could choose to not define it and rely on system fonts, or use a custom font. That way the component no longer directly relies on your brand design tokens.

Reply:
This is where the border between brand design tokens and design system tokens gets vague for me.

The idea of the `--denhaag-font-family` design token is that it's the default token for all font's in the design system, not just a token that only The Hague uses. Although I agree it will be useful to have layered tokens that get gradually get more specific. `--denhaag-font-family` > `--denhaag-form-input-font-family` > `--denhaag-button-font-family`.

Reply:
Yeah, so for Utrecht we now have three directories for design tokens:

- `brand/`
- `common/`
- `component/`

The `--denhaag-font-family` sibling there is `--utrecht-document-font-family` inside `common/`:

https://github.com/nl-design-system/utrecht/blob/336a6f1b9e6c9dc9305381327ae99404ae9601ef/proprietary/design-tokens/common/document.css#L7-L9

You can see it as default value for `paragraph` here, and we will be using it as fallback value for probably all rich text block level elements, but perhaps not for "native" elements such as `<select>`.

https://github.com/nl-design-system/utrecht/blob/240605699bea719d32a7cc68e35f49c99da04c17/src/paragraph/index.css#L6-L7

Reply:
So about the brand design tokens:

- `brand/` contains brand fonts, the brand color palette etc. It can contain fancy brand specific names such as "klm-skyblue" or `military-green` or something like that. Values in this level are not likely to change for a dark mode for example, or for a specific theme or a responsive version.
- `common/` and `component/` is still an expression of your brand, it just applies design decisions and brand values to the design system and the components. You might have different themes for this even for one brand, such as light-on-dark and dark-on-light, dark theme, responsive typography etc.

The values from `brand/` are likely managed by a marketing team with will likely have a different governance process than the application of those values in `common/` and `component/`.

These design tokens together are the theme for The Hague.

Why is brand a separate thing then? First of all, to encourage limiting the choices to stay within your brand parameters and to make it obvious you are going outside of the defined boundaries of your brand when you use a custom hex-color code for example. Second of all, because these brand design tokens can be useful for many projects in your organisation, and it is important to have a single source of truth. Every The Hague project should be able to use those tokens, regardless of wether they use NL Design System or your Material-UI components.

Eventually the design tokens can be made available in other formats than CSS inside a root selector, for that purpose of applying the tokens widely and serving as a single source of truth. For that we are exploring the use of [Style Dictionary](https://amzn.github.io/style-dictionary/) JSON files, you can get an impression of that in the new [amsterdam-design-tokens](https://github.com/nl-design-system/amsterdam-design-tokens/tree/main/proprietary/design-tokens/src) repository where we just started experimenting.

Reply:
We've implemented the separation between proprietary, common and component tokens. The hierarchical token structure will be implemented separately when we've added the design for more input components. (Trying to lock this PR down ;P)